README.md

```diff
@@ -151,27 +155,38 @@ module "aws_account" {
 }
 ```
 
-## IAM Permissions Boundaries
+## Permissions boundaries for IAM entities
+
+A permissions boundary is an advanced feature for using a managed policy to set the maximum permissions that an identity-based policy can grant to an IAM entity. An entity's permissions boundary allows it to perform only the actions that are allowed by both its identity-based policies and its permissions boundaries, such as preventing Terraform from creating a user or role with elevated privileges that an end-user could use to gain extra access.
+
+[See the AWS documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_boundaries.html) to learn more.
+
+This module supports attaching a permissions boundary on the role or user used by the workspace (`workspace_` prefix), as well as attaching a permission boundary to any created IAM role or user (`workload_` prefix).
+
+> **Note**
+> `workspace_boundary` and `workload_boundary` will be parsed by `templatefile`, so you can pass a template containing `account_id` to use the AWS account ID in your boundary policy.
 
-The module supports setting a Permission Boundary on the workspace `iam_user` or `iam_role` by passing down `permissions_boundaries.workspace_boundary`, which needs to be referencing the path where the permissions boundary is stored in git and the name: `permissions_boundaries.workspace_boundary_name`.
+### Workspace permissions boundary
 
-In case you want to reference a permission boundary that needs to be attached to every IAM role/user that will be created by the workspace role/user then you can create this permission boundary by specifying `permissions_boundaries.workload_boundary` which needs to be referencing the path where the permissions boundary is stored in git and the name: `permissions_boundaries.workload_boundary_name`.
+To create an attach a permissions boundary to the role or user used by the default workspace provider, specify the path to the policy and the desired policy name:
 
 ```hcl
-module "aws_account" {
-  source = "github.com/schubergphilis/terraform-aws-mcaf-avm?ref=VERSION"
-  ...
   permissions_boundaries = {
-    workspace_boundary      = "${path.module}/workspace_boundary.json"
+    workspace_boundary      = "${path.module}/policies/workspace_boundary.json"
     workspace_boundary_name = "workspace_boundary"
-    workload_boundary       = "${path.module}/workload_boundary.json"
-    workload_boundary_name  = "workload_boundary"
   }
-  ...
-}
 ```
 
-Note: the `workspace_boundary` and `workload_boundary` can be templated files, `account_id` will be replaced by AVM by the account ID of the AWS account created.
+### IAM resource permissions boundaries
+
+Similar to how a workspace permissions boundary is set, you can also attach a permissions boundary to any IAM role or user that is created by the workspace:
```

Comment:
Note: In order to enforce using the workload_boundary the workspace_boundary needs to deny creation of IAM resources unless the workload_boundary is attached. This can be attained (for example) by adding the following statements to the workload_boundary:

```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "DenyPermBoundaryIAMPolicyAlteration",
            "Effect": "Deny",
            "Action": [
                "iam:DeletePolicy",
                "iam:DeletePolicyVersion",
                "iam:CreatePolicyVersion",
                "iam:SetDefaultPolicyVersion"
            ],
            "Resource": [
                "arn:aws:iam::${account_id}:policy/platform/workload_boundary",
                "arn:aws:iam::${account_id}:policy/platform/workspace_boundary"
            ]
        },
        {
            "Sid": "DenyRemovalOfPermBoundaryFromAnyUserOrRole",
            "Effect": "Deny",
            "Action": [
                "iam:DeleteUserPermissionsBoundary",
                "iam:DeleteRolePermissionsBoundary"
            ],
            "Resource": [
                "arn:aws:iam::${account_id}:user/*",
                "arn:aws:iam::${account_id}:role/*"
            ],
            "Condition": {
                "StringEquals": {
                    "iam:PermissionsBoundary": "arn:aws:iam::${account_id}:policy/platform/workload_boundary"
                }
            }
        },
        {
            "Sid": "DenyAccessIfRequiredPermBoundaryIsNotBeingApplied",
            "Effect": "Deny",
            "Action": [
                "iam:PutUserPermissionsBoundary",
                "iam:PutRolePermissionsBoundary"
            ],
            "Resource": [
                "arn:aws:iam::${account_id}:user/*",
                "arn:aws:iam::${account_id}:role/*"
            ],
            "Condition": {
                "StringNotEquals": {
                    "iam:PermissionsBoundary": "arn:aws:iam::${account_id}:policy/platform/workload_boundary"
                }
            }
        },
        {
            "Sid": "DenyUserAndRoleCreationWithOutPermBoundary",
            "Effect": "Deny",
            "Action": [
                "iam:CreateUser",
                "iam:CreateRole",
                "iam:UpdateUser",
                "iam:UpdateRole",
                "iam:DeleteUserPolicy",
                "iam:AttachUserPolicy",
                "iam:DetachUserPolicy"
            ],
            "Resource": [
                "arn:aws:iam::${account_id}:user/*",
                "arn:aws:iam::${account_id}:role/*"
            ],
            "Condition": {
                "StringNotEquals": {
                    "iam:PermissionsBoundary": "arn:aws:iam::${account_id}:policy/platform/workload_boundary"
                }
            }
        }
    ]
}
```